### PR TITLE
Update react-loadable types for v4.

### PR DESCRIPTION
**Pull request by odensc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/thejameskyle/react-loadable/pull/46
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/react-loadable/index.d.ts [hunk 1]
```diff
@@ -24,64 +25,76 @@ export interface CommonOptions {
      * (this is considered a valid React component).
      */
     // NOTE: () => null is only needed until React.SFC supports components returning null
-    LoadingComponent: React.ComponentClass<LoadingComponentProps> | React.SFC<LoadingComponentProps> | (() => null);
+    loading: React.ComponentClass<LoadingComponentProps> | React.SFC<LoadingComponentProps> | (() => null);
```
Comment (Kovensky) [inline types/react-loadable/index.d.ts]: Same here, can be `React.ComponentType<LoadingComponentProps>`. Not sure if the `(() => null)` is still needed, but if `loading: () => null` doesn't pass without it then it is.

File: types/react-loadable/index.d.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-// Type definitions for react-loadable 3.3
+// Type definitions for react-loadable 4.0
 // Project: https://github.com/thejameskyle/react-loadable#readme
-// Definitions by: Diogo Franco <https://github.com/Kovensky>
+// Definitions by: Diogo Franco <https://github.com/Kovensky>, Oden S. <https://github.com/odensc>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.3
 
 import * as React from 'react';
```
Comment (Kovensky) [inline types/react-loadable/index.d.ts]: The `LoadedComponent` type itself can now just be an alias to `React.ComponentType`, which didn't exist at the time this was written.

Or we could take the opportunity to just remove it and use `React.ComponentType`; this update is a breaking change anyway.

**Review by Jessidhia — APPROVED** [review]
Looks good, though this is a good opportunity to switch to using the `React.ComponentType` alias instead of the local `LoadedComponent` alias.

File: types/react-loadable/index.d.ts (hunk 1 above)
Comment (odensc) [inline types/react-loadable/index.d.ts]: Thanks @Kovensky - I wasn't aware of `React.ComponentType`. I made that change (and a couple other minor ones), would you mind taking another look when you have the chance?

Comment (Kovensky) [inline types/react-loadable/index.d.ts]: Just need to change this to `React.ComponentType<LoadingComponentProps> | (() => null)`

**Review by Jessidhia — APPROVED** [review]

**Review by Jessidhia — APPROVED** [review]

**Comment by rohmanhm** [comment]
This type not yet available via `typings search react-loadable`.

**Pull request merged**